### DISCUSSION TITLE
Reduce Jellyfin RAM / memory usage

### DISCUSSION THREAD

**Discussion by ngosang** [jellyfin/jellyfin#10444] [General] [answered]
I'm coming form [this issue](https://github.com/jellyfin/jellyfin/issues/6306) where more than 20 users complained about Jellyfin crashing, using much RAM or a possible memory leak.

There is not such memory leak since the memory is managed by .Net garbage collector and the [memory requirements are 8GB of RAM](https://jellyfin.org/docs/general/administration/hardware-selection/) and most users in that issue are running in low-end hardware without enough resources.

Said that, I want to start a discussion about **how we can reduce the minimum memory from 8GB -> 500 MB**. This will allow to run Jellyfin in low-end hardware (like SBC, Raspberry Pi, NAS...) and make it more popular.

I think it is technically possible with small changes in the current code. How do I know?

1. I use Radarr, Sonarr, Lidarr, Jackett. They are all written in .Net and manage the same media library as Jellyfin. They use the same files, the same posters, same metadata, have a nice web ui so they are comparable to Jellyfin. IMHO they do more complex things than Jellyfin with a lot of background tasks and interactions with 3rd party APIs. Those apps use between 100 and 300 MB of RAM. Never more than 300MB even if they run for months. Jellyfin uses about 5GB with the same media library. I have all transcoding disabled.
2. [As I have commented here.](https://github.com/jellyfin/jellyfin/issues/6306#issuecomment-1107823711) I can start Jellyfin, visit the main page, select a movie and watch it all using only 400 MB of RAM (no transcoding). When I start browsing the media library and days go by, the memory increases to 5 GB. This tells me that the main functionality needs <500 MB to run without changes in the code. The rest of the memory is being used to cache things that are not important. If instead of having them in memory, they are queried in the database, the response time will be a few milliseconds slower, but Jellyfin could run on many more devices.

**Comment by oddstr13** [comment]
My Jellyfin instance is using 2GB RAM, in a VM with 4GB allocated RAM.
Jellyfin is the main user of memory on this box, so the memory pressure is fairly low.

![image](https://github.com/jellyfin/jellyfin/assets/638706/748bef95-fef6-454f-bc46-535a66611d1b)
Seen here is memory usage the past year, along with .net version which here serves as proxy for Jellyfin update.
Dips in the graph represents restarts, the cause of which is *usually* one of;
- System taken down as a precautionary measure during thunderstorm
- Updates, either to the system, Jellyfin or plugins

**Comment by ngosang** [comment, marked as answer]
I have Jellyfin running in 250MB of RAM => https://github.com/jellyfin/jellyfin/issues/6306#issuecomment-1774093928
